it/src/test/java/com/google/cloud/teleport/it/bigtable/DefaultBigtableResourceManagerTest.java

```diff
@@ -0,0 +1,392 @@
+    prepareCreateInstanceAdminClientMock();
+    prepareCreateTableAdminClientMock();
+    prepareCreateDataClientMock();
+    testManager.createInstance(cluster);
+
+    when(bigtableTableAdminClient.exists(anyString())).thenReturn(true);
+    when(bigtableDataClient.readRows(any())).thenReturn(null);
+
+    assertThrows(NullPointerException.class, () -> testManager.readTable(TABLE_ID));
+  }
+
+  @Test
+  public void testReadTableShouldWorkWhenBigtableDoesNotThrowAnyError() throws IOException {
+    prepareCreateInstanceAdminClientMock();
+    prepareCreateTableAdminClientMock();
+    prepareCreateDataClientMock();
+    testManager.createInstance(cluster);
+
+    when(bigtableDataClient.readRows(any())).thenReturn(readRows);
+    when(bigtableTableAdminClient.exists(anyString())).thenReturn(true);
+
+    testManager.readTable(TABLE_ID);
+  }
+
+  @Test
+  public void testCleanupCallsDeleteInstance() {
+    prepareCreateInstanceAdminClientMock();
+
+    doThrow(RuntimeException.class).when(bigtableInstanceAdminClient).deleteInstance(anyString());
+
+    assertThrows(RuntimeException.class, () -> testManager.cleanup());
+  }
+
+  @Test
+  public void testCleanupClosesInstanceAdminClient() {
+    prepareCreateInstanceAdminClientMock();
+
+    doThrow(RuntimeException.class).when(bigtableInstanceAdminClient).close();
+
+    assertThrows(RuntimeException.class, () -> testManager.cleanup());
+  }
+
+  @Test
+  public void testCleanupClosesTableAdminClient() {
+    prepareCreateInstanceAdminClientMock();
+    prepareCreateTableAdminClientMock();
+
+    doThrow(RuntimeException.class).when(bigtableTableAdminClient).close();
+
+    assertThrows(RuntimeException.class, () -> testManager.cleanup());
+  }
+
+  @Test
+  public void testCleanupShouldThrowErrorWhenInstanceFailsToDelete() {
+    prepareCreateInstanceAdminClientMock();
+
+    doThrow(RuntimeException.class).when(bigtableInstanceAdminClient).deleteInstance(anyString());
+
+    assertThrows(BigtableResourceManagerException.class, () -> testManager.cleanup());
+  }
+
+  @Test
+  public void testCleanupShouldThrowErrorWhenInstanceAdminClientFailsToClose() {
+    prepareCreateInstanceAdminClientMock();
+
+    doThrow(RuntimeException.class).when(bigtableInstanceAdminClient).close();
+
+    assertThrows(BigtableResourceManagerException.class, () -> testManager.cleanup());
+  }
+
+  @Test
+  public void testCleanupShouldThrowErrorWhenTableAdminClientFailsToClose() {
+    prepareCreateInstanceAdminClientMock();
+    prepareCreateTableAdminClientMock();
+
+    doThrow(RuntimeException.class).when(bigtableTableAdminClient).close();
+
+    assertThrows(BigtableResourceManagerException.class, () -> testManager.cleanup());
+  }
+
+  @Test
+  public void testCleanupShouldWorkWhenBigtableDoesNotThrowAnyError() {
+    prepareCreateInstanceAdminClientMock();
+    prepareCreateTableAdminClientMock();
+
+    testManager.cleanup();
```

Comment:
Can we put some asserts/verify() to make sure it worked as planned?

Reply:
There are 3 other tests to make sure the intended resources throw errors when there is a problem, and therefore tests to make sure the 3 necessary close/delete functions are called. This test then subsequently ensures no errors are thrown when the 3 functions work as intended. Special logic could be implemented, but I feel as though that would be redundant in this case.

Reply:
Refactored to use the `Mockito.verify` method/check.